README.md

```diff
@@ -75,24 +75,55 @@ $ yarn build:docs
 ```
 
 ## Unit/Integration tests
-
+### Overview
 owncloud-sdk uses [pactjs](https://github.com/pact-foundation/pact-js) with jest for unit and integration test.
-Before running tests you need to create config.json file.
+
+On the pact provider side, tests have 4 different `interactions`:
+
+- interactions that work on both oc10 & ocis
+- interactions that are "pending" on oc10 but should work on ocis
+- interactions that are "pending" on ocis but should work on oc10
+- interactions that are "pending" on both ocis & oc10
+
+The CI is not expected to fail for the interactions that are "pending" but it's expected to fail for those interactions that were already verified and started failing. Pact.io has a system to handle such a scenario called [pending pacts](https://docs.pact.io/pact_broker/advanced_topics/pending_pacts/ 'pending pacts'). This feature allows changed contracts to be verified without failing the provider's build.
+
+Four different pacts for the different buckets of interactions are created when running the provider tests. Pacts that are allowed to fail are marked as `pending` else not.
+
+In a consumer test, a new provider is created using the function `createProvider`. It is needed to be called with two parameters, depending upon which provider-version the test is still pending.
+
+To add a new consumer test which is expected to fail on ocis but pass on oc10 provider, provider should be created as:
+
+```js
+describe('feature', function () {
+  it('new test feature', function () {
+    const isPendingOnOc10 = true // set this as true if the test is allowed to fail on oc10 provider
+    const isPendingOnOcis = false // set this as true if the test is allowed to fail on ocis provider
+    const provider = createProvider(isPendingOnOc10, isPendingOnOcis)
+    // get interactions
+    // execute test
+    provider.executeTest(() => {
+      // test body
+    })
+  })
+})
+```
+### Running Tests
+At first, you need to create `config.json` file.
 ```
 $ cp tests/config/config.sample.json tests/config/config.json
 ```
 
 ### consumer tests
-The pact consumer tests checks owncloud sdk against the pact mock server. In order to run these tests, use this comand
+The pact consumer tests checks owncloud sdk against the pact mock server. In order to run these tests, use the following command:
 
 ```
 $ yarn test-consumer
 ```
 
-If you have pacts from old test run in `tests/pacts` your tests will fail. Make sure to delete that before you run the tests again.
+> If you have pacts from old test run in `tests/pacts` your tests will fail. Make sure to delete that before you run the tests again.
```

Comment:
why this change?

Reply:
this is kind of a note rather than how to run.